Sources/App/Controllers/ImageController.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  ImageController.swift
+//  App
+//
+//  Created by Zach Eriksen on 10/19/19.
+//
+
+import Vapor
+import Imperial
+import Authentication
+
+struct ImageController: RouteCollection {
+    let imageFolder = "uploads/"
+    
+    func boot(router: Router) throws {
+        let authSessionRouter = router.grouped(User.authSessionsMiddleware())
+        
+        authSessionRouter.post("uploadImage", use: uploadImageHandler)
+        authSessionRouter.get("image", ImageUpload.parameter, use: getImageHandler)
+    }
+    
+    func uploadImageHandler(_ req: Request) throws -> Future<ImageUploadResponse> {
+        let user = try req.requireAuthenticated(User.self)
+        return try req.content
+            .decode(ImageUpload.self)
+            .flatMap { imageData in
+                let workPath = try req.make(DirectoryConfig.self).workDir
+                let name = try "\(user.requireID())-\(UUID().uuidString).jpg"
+                let path = workPath + self.imageFolder + name
+                FileManager().createFile(atPath: path,
+                                         contents: imageData.picture,
+                                         attributes: nil)
+                
+                return Future.map(on: req) { ImageUploadResponse(fileName: name) }
+        }
+    }
+    
+    func getImageHandler(_ req: Request) throws -> Future<ImageUpload> {
+        let imageName = try req.parameters.next(ImageUpload.self)
+        
+        let path = try req.make(DirectoryConfig.self).workDir + self.imageFolder + imageName
```

Comment:
looks like this code is reused, could either move it into a utility function, or set up the first part of the path in the init

Reply:
I can make it into a util function